zulipterminal/ui_tools/boxes.py

```diff
@@ -1200,16 +1200,16 @@ def keypress(self, size: urwid_Size, key: str) -> Optional[str]:
             #   ```quote
             #   message_content
             #   ```
-            quote = '@_**{}|{}** '.format(self.message['sender_full_name'],
-                                          self.message['sender_id'])
-
-            # FIXME: The type and generation of link needs to be discussed.
-            # For interoperability webapp link should be given, but
-            # that would not narrow to the quoted message inside ZT.
             absolute_url = absolute_url_of_message(self.model, self.message)
-            quote += '[said]({}):\n'.format(absolute_url)
-            quote += '```quote\n' + self.model.client.get_raw_message(
-                self.message['id'])['raw_content'] + '\n```\n'
+            fence = get_unused_fence(self.model.client.get_raw_message(
+                self.message['id'])['raw_content'])
+
+            quote = '@_**{0}|{1}** [said]({2}):\n{3}quote\n{4}\n{3}\n'.format(
+                        self.message['sender_full_name'],
+                        self.message['sender_id'], absolute_url, fence,
+                        self.model.client.get_raw_message(
+                            self.message['id'])['raw_content'])
```

Comment:
Separate lines for each parameter and extracting the raw message calculation would make this easier to read. We can use a smaller indent too?

This won't apply if you put this commit first, but when you have both the fencing and improving quoting, note that we want to avoid calling the server twice for the get_raw_message.